### PR TITLE
Add separate targetTemperature sensor on washing machines and washer-dryers.

### PR DESCRIPTION
**Pull request by aturri**
The targetTemperature (0) sensor was used for washing machines in the same way as for other appliances (fridges, ovens, etc.).

Create a separate sensor used only by washing machines and washer-dryers in order to display target temperature as an integer. Sensor key is still "targetTemperature" so it's not a breaking change.

Target temperature 1 and 2 are not in use by any model of these classes.

### CODE REVIEW

File: custom_components/miele/sensor.py [hunk 1]
```diff
@@ -174,14 +174,29 @@ class MieleSensorDefinition:
...
+        ],
+        description=MieleSensorDescription(
+            key="targetTemperature",
+            data_tag="state|targetTemperature|0|value_raw",
+            device_class=SensorDeviceClass.TEMPERATURE,
+            name="Target temperature",
+            native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+            state_class=SensorStateClass.MEASUREMENT,
+            entity_category=EntityCategory.DIAGNOSTIC,
+            decimals=0,
```
Comment (astrandb) [inline custom_components/miele/sensor.py:187]:
```suggestion

```
Decimals is a leftover from cut&paste from another integration. Has no function in this integration. Should be removed.

File: custom_components/miele/sensor.py [hunk 2]
```diff
@@ -174,14 +174,29 @@ class MieleSensorDefinition:
...
             WASHING_MACHINE,
+            WASHER_DRYER,
+        ],
+        description=MieleSensorDescription(
+            key="targetTemperature",
+            data_tag="state|targetTemperature|0|value_raw",
+            device_class=SensorDeviceClass.TEMPERATURE,
+            name="Target temperature",
+            native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+            state_class=SensorStateClass.MEASUREMENT,
```
Comment (astrandb) [inline custom_components/miele/sensor.py:185]:
```suggestion

```
This sensor is not a measurement according to the spec in https://developers.home-assistant.io/docs/core/entity/sensor#available-state-classes. It should be removed from other targetTemperature sensors also. I can do it in another PR.

**Review by astrandb — COMMENTED** [review]
If I understand you correctly the intention with this PR is to display target temperature without decimals and to remove two hidden target temperatures.
I think the target without decimals is a valid change for all devices with target temperature.
It was a practical solution to create deactivated sensors for appliance types with different number of zones. There is no simple way do deduct from the API how many zones there are for a specific device, except by creating and maintaining specific capabilities tables for each device model. If you are sure that there is no and never will be any multizone devices of type washer or dryer, well then it is OK to remove zone 2 and zone3

**Comment by aturri** [comment]
> If I understand you correctly the intention with this PR is to display target temperature without decimals and to remove two hidden target temperatures.

Yeah, you understood correctly.

> I think the target without decimals is a valid change for all devices with target temperature.

Is this true also for fridges / wine cabinets? I don't have these device classes, so I can't verify this. In that case, this change could be applied also to zones 2 and 3 for all device classes.

>  It was a practical solution to create deactivated sensors for appliance types with different number of zones. There is no simple way do deduct from the API how many zones there are for a specific device, except by creating and maintaining specific capabilities tables for each device model. If you are sure that there is no and never will be any multizone devices of type washer or dryer, well then it is OK to remove zone 2 and zone3

Of course it's an assumption that zones 2 and 3 will never be used by washers / washer-dryers, but it seems reasonable. It's not a problem to revert this and leave deactivated sensors, it's more generic and practical.

**Comment by astrandb** [comment]
> > I think the target without decimals is a valid change for all devices with target temperature.
> 
> Is this true also for fridges / wine cabinets? I don't have these device classes, so I can't verify this. In that case, this change could be applied also to zones 2 and 3 for all device classes.

The API data model specifies targetTemperature as an integer when setting the target, so I think it is reasonable do display the returned value without decimals. I can confirm that fridge and freezer only can be set on integer temparatures on the panel.

I suggest that the received targetTemperature is converted to an int for all instances.
If we should separate washers from the other devices with targetTemperatures when there are no material differences is a matter of opinion. You can decide!
And when you are in the sensor.py file. Please remove `state_class=SensorStateClass.MEASUREMENT` from all targetTemperature sensors.

**Comment by aturri** [comment]
Very well, so I'll update this also for other target temperature sensors.

> If we should separate washers from the other devices with targetTemperatures when there are no material differences is a matter of opinion. You can decide!

So, probably it's better previous solution with deactivated sensors... so if something changes on Miele side, it's easier to maintain because other 2 sensors are already available.

**Comment by astrandb** [comment]
Thank you!

**Pull request merged**